### ISSUE TITLE
Strange output while training 

### ISSUE THREAD

**Issue by SJRogue** [AlexeyAB/darknet#371]
Hello Alexey, Nils, anybody else involved enough,

I am getting the following freak output after about 130 iterations. Not sure where I went wrong.
I dug through the code but I am no C/C++ developer so "nan" and "-nan(ind)" are not so clear to me.

When I can help anyone else I will.. : )

[EDIT:
From what I've read so far the -nan and -nan(ind)  output is because I could be exceeding the precision of float primitives but.. this would seem strange to me.

I could go change all the floats to doubles but this does not seem like a good solution to start changing the framework so deeply]

![afbeelding](https://user-images.githubusercontent.com/10557456/35781688-a0b976ee-09ed-11e8-8805-6cd784462d49.png)

**Comment by AlexeyAB** [comment]
If all values is `nan` then training goes wrong.

* Sometimes, it happens even if you do everything right. **Just run training from the begining**.
* Also it can happen if there is used wrong labels in the training dataset, or wrong cfg-file: 
    * if batch to small 
    * or used worng policy and learning_rate to small

**Comment by SinghJorawar** [comment]
Thanks again Alexey,

I saw indeed that when i started from the beginning with batch size back to 64 as you said and more normal parameters, the training is succeeding well.

This is going good.. my avg loss error is dropping immensely with each iteration.
 2.708600 avg after 33 iterations.
